presto-mysql/src/test/java/io/prestosql/plugin/mysql/TestingMySqlServer.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.plugin.mysql;
+
+import org.testcontainers.containers.MySQLContainer;
+
+import java.io.Closeable;
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.Properties;
+
+import static java.lang.String.format;
+import static org.testcontainers.containers.MySQLContainer.MYSQL_PORT;
+
+public class TestingMySqlServer
+        implements Closeable
+{
+    private final MySQLContainer<?> dockerContainer;
+
+    public TestingMySqlServer()
+    {
+        this.dockerContainer = new MySQLContainer<>("mysql:8.0.15");
+        dockerContainer.withDatabaseName("tpch");
+        dockerContainer.start();
+        execute(format("GRANT ALL PRIVILEGES ON *.* TO '%s'", dockerContainer.getUsername()), "root", dockerContainer.getPassword());
+    }
+
+    public String getJdbcUrl()
+    {
+        return format("jdbc:mysql://%s:%s?useSSL=false&allowPublicKeyRetrieval=true", dockerContainer.getContainerIpAddress(), dockerContainer.getMappedPort(MYSQL_PORT));
```

Comment:
Use `getJdbcUrl()` from `MySQLContainer` which already adds these options

Reply:
I misread the code. This is done in the protected method `constructUrlForConnection()`.